pilot/pkg/proxy/envoy/v2/cds.go

```diff
@@ -78,7 +78,14 @@ func (s *DiscoveryServer) generateRawClusters(node *model.Proxy, push *model.Pus
 		return nil, err
 	}
 
+	if sdsTokenPath, found := node.Metadata[model.NodeMetadataSdsTokenPath]; found && len(sdsTokenPath) > 0 {
+		// If SDS_TOKEN_PATh is in the node metadata, make a copy of rawClusters so that
+		// the path of SDS token will be applied to the copied clusters.
+		rawClusters = CopyClusters(rawClusters)
```

Comment:
Will the original rawClusters lose its reference if you overwrite with its copy?

Reply:
No, rawClusters won't lose its reference.